frontend/src/type-interfaces/FieldEditState.tsx

```diff
@@ -0,0 +1,6 @@
+export interface FieldEditState {
+  name: boolean,
+  email: boolean,
+  major: boolean,
+  courses: boolean,
+}
```

Comment:
Cool, but possibly overkill? At least as something in the store. Could use in local state instead